zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/BackupStorageProvider.java

```diff
@@ -95,4 +95,15 @@
    * @throws IOException
    */
   void cleanupInvalidFiles(File path) throws IOException;
+
+  /**
+   * Copy the selected snapshot and transaction log backup files needed for restoration to a temp folder;
+   * Truncate the copied transaction log to zxid restoration point.
+   * @param tempDirDestination The path to create a temporary directory and store the copied backup files
+   * @param filesToCopy A list of selected backup files that can be used for restoration
+   * @param zxidToRestore The zxid to restore to
+   * @throws IOException
+   */
+  void processBackupFilesForRestoration(File tempDirDestination, List<BackupFileInfo> filesToCopy,
```

Comment:
I'm not sure if this is the right place for this processing logic (interface) to be in. In my opinion, Storage Provider interface should strictly provide I/O interface methods - restoration logic should go to a different place - either a separate interface, or part of the restoration tool? Do you think we should move this logic elsewhere?